enhancements/microshift/gateway-api-support.md

```diff
@@ -0,0 +1,434 @@
+and [Kubernetes ingress](https://docs.redhat.com/en/documentation/red_hat_build_of_microshift/4.16/html/networking/microshift-configuring-routes#nw-ingress-creating-a-route-via-an-ingress_microshift-configuring-routes).
+In order to add more routing features a new project was born in upstream:
+[Gateway API](https://gateway-api.sigs.k8s.io/). This is an official kubernetes
+project focused on L7 routing (support for L4 is experimental), posed to be the
+next standard in ingress, load balancer and service mesh APIs.
+
+This enhancement proposes an implementation to support this new API as an
+optional component in MicroShift.
+
+## Motivation
+Gateway API support will provide users with another way of expressing their
+ingress routing capabilities, also aligned with the way Kubernetes foresees
+this functionality.
+
+### User Stories
+* As a MicroShift admin, I want to add different Gateway resources to the
+  cluster for applications to create routes on them.
+* As a MicroShift application developer, I want to add HTTP routes to specific
+  Gateways so that I can express routing capabilities using an upstream API.
+* As a MicroShift application developer, I want to add GRPC routes to specific
+  Gateways so that I can express routing capabilities using an upstream API.
+
+### Goals
+- Full support for core, non-experimental resources, including gateways, HTTP routes and GRPC routes.
+- Full support for multiple Gateway instances.
+- Full support for north/south [use case](https://gateway-api.sigs.k8s.io/concepts/use-cases/#basic-northsouth-use-case).
+- Lightweight implementation, consuming as little resources as possible.
+
+### Non-Goals
+- Automatically removing Gateway API from the cluster upon RPM uninstall
+- Provide support for experimental resources, such as those listed [here](https://gateway-api.sigs.k8s.io/concepts/api-overview/#route-resources).
+- Support different GatewayClasses with the same controller.
+
+## Proposal
+As the Gateway API is still in development and MicroShift already provides
+means for routing capabilities, this feature should be optional.
+
+As with other optional features there will be an optional RPM with all the
+required manifests to be applied upon MicroShift's start. The backing
+implementation for Gateway API support will be based on the latest OpenShift
+Service Mesh component, which is an optional operator from OpenShift.
+
+OpenShift Service Mesh (OSSM) is the service mesh operator from OpenShift.
+Since version 3 (which is going to be Tech Preview at the time of this writing)
+it is based in the [sail operator](https://github.com/openshift-service-mesh/sail-operator/tree/main),
+and is released as an optional operator in the Red Hat operators catalog for
+OLM. This operator deploys all of Istio's manifests and is capable of handling
+Istio control plane deployments, Istio CNIs and other components. Istio has
+first class support for Gateway API resources and is already supported by Red
```

Comment:
A definition of "first class support" from Istio for Gateway API resources is needed here.  Again, whenever you use the term "support" you are setting up a big expectation.

Reply:
Rephrased it.